docs-en/chapter_computational_complexity/iteration_and_recursion.md

```diff
@@ -1,194 +1,199 @@
-# Iteration vs. Recursion
+# Iteration and Recursion
 
-In data structures and algorithms, it is common to repeat a task, which is closely related to the complexity of the algorithm. There are two basic program structures that we usually use to repeat a task: iteration and recursion.
+In algorithms, repeatedly performing a task is quite common and closely linked to complexity analysis. Therefore, before diving into the details of time complexity and space complexity, let's first understand how to implement repetitive tasks in programming, focusing on two fundamental control structures: iteration and recursion.
 
 ## Iteration
 
-An "iteration iteration" is a control structure that repeats a task. In iteration, a program repeats the execution of a piece of code until the condition is no longer satisfied.
+"Iteration" is a control structure used to repeatedly execute a task. In an iterative process, a program executes a block of code repeatedly until certain conditions are met, ceasing once this condition is no longer satisfied.
 
 ### For Loops
 
-`for` loops are one of the most common forms of iteration, **suitable when the number of iterations is known in advance**.
+The `for` loop is one of the most common forms of iteration, and **it's particularly suitable when the number of iterations is known in advance**.
 
-The following function implements the summation $1 + 2 + \dots + n$ based on a `for` loop, and the result is recorded using the variable `res`. Note that `range(a, b)` in Python corresponds to a "left-closed-right-open" interval, which is traversed in the range $a, a + 1, \dots, b-1$.
+The following function implements summation $1 + 2 + \dots + n$ using a `for` loop, with the sum being recorded in the variable `res`. Note that in Python, `range(a, b)` creates a range from $a$ up to but not including $b$, making it an exclusive range. This means it includes numbers starting from $a$ up to $b−1$, but excludes $b$ itself.
 
 ```src
 [file]{iteration}-[class]{}-[func]{for_loop}
 ```
 
-The figure below shows the flow block diagram of this summation function.
+Figure 2-1 shows the flowchart for this summation function.
 
 ![Flow block diagram of the summation function](iteration_and_recursion.assets/iteration.png)
+<div align="center">Figure 2-1: Flowchart of the Summation Function</div>
```

Comment:
The caption will be added in the building process, so we don't have to add it here. Same as others.